### ISSUE TITLE
Global Illumination and lights.

### ISSUE THREAD

**Issue by elfl0rd** [StrangeLoopGames/EcoIssues#14766]
0.9 with new setings is not fixed this old issue...
Just look
GI - off
ceiling light and street light - too less light.
![20200215212830_1](https://user-images.githubusercontent.com/4980243/74593244-51dd3c00-503a-11ea-8c0a-09079a90e298.jpg)

GI - on
ceiling light ok, all other lamps - bad.
![20200215212839_1](https://user-images.githubusercontent.com/4980243/74593248-586bb380-503a-11ea-963e-bdf70d8ad12f.jpg)

Also ceiling light have light source at center of model, i think its wrong.

**Comment by SlayksWood** [comment]
From #15466
> Version [0.9.0 staging-1489]
> Street Lamp and Electric Wall Lamp
> Both lamps generate way too low light is almost inexistent,moon generated allot more light then them

**Comment by SlayksWood** [comment]
> **Version**: 0.9.0 staging-1554
> **Steps to reproduce**:
> 
> 1. Be indoor
> 2. Put some braziers and/or fireplaces in the room
> 3. Wait until nighttime
> 4. Switch between GI off and low,medium,high,ultra
> 
> GPU: RTX 2070 Super
> 
> GI off:
> ![GI_OFF](https://user-images.githubusercontent.com/39825431/82384590-7af1dc00-9a30-11ea-88ba-c82be7c4d277.jpg)
> GI low:
> ![GI_LOW](https://user-images.githubusercontent.com/39825431/82384603-80e7bd00-9a30-11ea-9bcc-49e26081f3c8.jpg)
> GI medium:
> ![GI_MEDIUM](https://user-images.githubusercontent.com/39825431/82384609-8513da80-9a30-11ea-8b9f-86a828c8f972.jpg)
> GI high:
> ![GI_HIGH](https://user-images.githubusercontent.com/39825431/82384622-880ecb00-9a30-11ea-8be1-681618e76a08.jpg)
> GI ultra:
> ![GI_ULTRA](https://user-images.githubusercontent.com/39825431/82384632-8cd37f00-9a30-11ea-90fe-2e8ccbb0ec39.jpg)
> 
> My other settings:
> ![Graphics Settings](https://user-images.githubusercontent.com/39825431/82384641-90ff9c80-9a30-11ea-9499-5a44a7246898.jpg)

**Comment by keegan-o** [comment]
I don't know how it will be possible to reconcile such radically different looks on different systems with the same settings-- something is causing GI to display very differently on Kirill's machine than on mine @redwyre

**Comment by keegan-o** [comment]
For me, having GI on is always brighter than off

**Comment by keegan-o** [comment]
Somehow, all the links to meshes on the GI objects in the prefabs got broken.... like all of them.

**Comment by keegan-o** [comment]
I went through and committed some kind of GI cube to all light emitting objects, but unfortunately some will probably end up being too small to work properly-- I have been editing them and the materials they use as I go and most are fixed again. I'm going to show phlo how to do this so he can give proper attention to the lighting objects; he is going to be adding new ones soon as well.

**Comment by Phlo123** [comment]
I've modified all GI objects, added new materials and meshes for existing lights to better reflect their size and light intensity.
All GI materials have their light type (electrics or fire) as well as intensity in their name.

**Comment by beastyn** [comment]
I still don't see light from the sources with GI Low and Medium. I'm not sure if this is about GI itself or whatever. But it still incomplete anyway. Push it for later releases (for @keegan-o  to review).
No GI
![bandicam 2020-09-02 12-54-58-919](https://user-images.githubusercontent.com/27898520/91955053-0f502800-ed1c-11ea-92f1-ac75d0fd1c4b.jpg)

Low GI
![bandicam 2020-09-02 12-55-10-894](https://user-images.githubusercontent.com/27898520/91955336-7241bf00-ed1c-11ea-96b1-9eee5ee163ee.jpg)
Medium GI
![bandicam 2020-09-02 12-55-19-976](https://user-images.githubusercontent.com/27898520/91955348-78d03680-ed1c-11ea-80f5-2c63e4803c1a.jpg)

**Comment by ThePiachu** [comment]
Adding to this - with Global Illumination the lights don't add much brightness to the game. Without Global Illumination, they help a bit, but not as much as I would expect something like a Street Lamp or Steel Ceiling Light in a closed room to add. I want to be able to put in 2-3 of those in my factory and for it to be as bright as daylight, rather than just machines getting some shine to them

![image](https://user-images.githubusercontent.com/964559/95666699-facc3000-0b10-11eb-9710-56918841fd77.png)

**Comment by keegan-o** [comment]
Basically the problem here is that altering the GI settings to be performant or different in any way means that the amount of light different GI objects emits changes-- currently, it means some settings work and others do not. This is a big extensibility/flexibility issue with the SEGI GI system we're using, annoying enough that we're going to focus on removing that system from being required for the lights to look ok with it turned on. We will likely just disable the functionality that disables the regular lights when you turn GI on, and go from there. That way at least everything that is supposed to will always emit light and we can figure out a simpler way for the GI system to also grant a bonus to the look, probably by making everything have a nested prefab GI light object rather than tons of different settings for every GI object.

**Comment by keegan-o** [comment]
I did find a quick fix that seemed to re-enable light from GI sources on medium and low settings which I commited to develop/staging/release, so that is a good bit of duct tape on this for now. Long term I'm going to be removing SEGI GI from lighting objects, and refining the actual lights on them to be lower range but higher intensity, with effects added here and there like volumetric beams. I've done this so far for the two street lights.
